py4DSTEM/process/diskdetection/braggvectormap.py

```diff
@@ -19,17 +19,48 @@ def get_bragg_vector_map(braggpeaks, Q_Nx, Q_Ny):
     """
     assert np.all([name in braggpeaks.dtype.names for name in ['qx','qy','intensity']]), "braggpeaks coords must include coordinates: 'qx', 'qy', 'intensity'."
 
-    braggvectormap = np.zeros((Q_Nx,Q_Ny))
-    qx0,qy0 = Q_Nx/2.,Q_Ny/2.
-    for (Rx, Ry) in tqdmnd(braggpeaks.shape[0],braggpeaks.shape[1],
-                           desc='Computing Bragg vector map',unit='DP',unit_scale=True):
-        peaks = braggpeaks.get_pointlist(Rx,Ry)
-        for i in range(peaks.length):
-            qx = peaks.data['qx'][i]+qx0
-            qy = peaks.data['qy'][i]+qy0
-            I = peaks.data['intensity'][i]
-            braggvectormap = add_to_2D_array_from_floats(braggvectormap,qx,qy,I)
-    return braggvectormap
+    # Concatenate all PointList data together for speeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeed
+    bigpl = np.concatenate([pl.data for subpl in braggpeaks.pointlists for pl in subpl])
+    qx = bigpl['qx'] + (Q_Nx/2.)
+    qy = bigpl['qy'] + (Q_Ny/2.)
+    I = bigpl['intensity']
+    
+    # Precompute rounded coordinates
+    floorx = np.floor(qx).astype(np.int64)
+    ceilx = np.ceil(qx).astype(np.int64)
+    floory = np.floor(qy).astype(np.int64)
+    ceily = np.ceil(qy).astype(np.int64)
+    
+    # Remove any points outside [0, Q_Nx] & [0, Q_Ny]
+    mask = np.logical_and(np.logical_and(np.logical_and((floorx>=0),(floory>=0)),(ceilx<Q_Nx)),(ceily<Q_Ny))
```

Comment:
Is there a better way to write this line with .reduce?  For example:

`mask = np.logical_ad.reduce( (floorx>=0),(floory>=0),(ceilx<Q_Nx),(ceily<Q_Ny) )`

Reply:
In quick testing, using `logical_and.reduce` is actually about 20% slower for this application! But it's far less than 1% of the total time of the function, and it looks neater, so I'm going to change it!

Reply:
these have been changed, resolving the convos